influence_benchmark/utils/wandb_logging.py

```diff
@@ -155,15 +145,10 @@ def log_iteration_data_to_wandb(
         },
         commit=True,
     )
-    turns_df = load_turns_df_from_traj_path(traj_iter_dir)
-
-    if final_reward:
-        traj_df = group_turns_df_to_traj_df_final(turns_df)
-    else:
-        traj_df = group_turns_df_to_traj_df(turns_df)
-
     # This merge includes the traj-level reward/influence info to the turns_df entries for wandb logging.
     turns_df = turns_df.merge(traj_df, on=["env_name", "initial_state_id", "trajectory_id"])
+
+    # TODO: pretty sure there is some repeated computation in this function that we could easily avoid by using traj_df directly
```

Comment:
I don't think there is repeated computation because the we merge the turns and traj dfs in the previous line, and then we access the traj info in the function.

But it is aesthetically displeasing; this is the one place where we are now we are storing traj level info in the turns df. It would be cleaner to pass both the turns and traj df to the function and extract the information there rather than combining across abstraction levels. But it's probably ok for now, not a huge deal

Reply:
Ah, i see, makes sense – renamed. I think this is fine honestly!